ddl/ddl_api.go

```diff
@@ -616,6 +616,14 @@ func columnDefToCol(ctx sessionctx.Context, offset int, colDef *ast.ColumnDef, o
 		col.Flag &= ^mysql.BinaryFlag
 		col.Flag |= mysql.ZerofillFlag
 	}
+	// Trailing spaces are automatically deleted from SET/ENUM member values in the table definition when a table is created.
+	// https://dev.mysql.com/doc/refman/8.0/en/enum.html#enum-literals
+	// https://dev.mysql.com/doc/refman/8.0/en/set.html
+	if col.Tp == mysql.TypeEnum || col.Tp == mysql.TypeSet {
+		for i := range col.Elems {
+			col.Elems[i] = collate.TruncateTailingSpace(col.Elems[i])
+		}
+	}
```

Comment:
Do we need to do this logic in `processColumnOptions` which use in `alter table... change column ...`？
We can add the test of `alter table t change a aa enum('1 ', '1\t', ' 1 ');`  to `TestIssue18949`.

Reply:
you are right, it is better trim trailing spaces in parser. I going to revert this PR.   This PR should only add tests for it